File: 12-2-2024_Recursive_sequence.cpp
class Solution{
public:
    long long sequence(int n){
        // code here
        long long int i=1,k,c=1,s=1;
        long long int p=0;
        long long mod=1000000007;
        while(i<=n)
        {
        k=i;
        while(k--)
        {
            s=(s*c)%mod;
            c++;   
        }
        
        p=p+s;
        p=p%mod;
        s=1;
        i++;
        }
        return p;
        }
};

File: 12-2-2024_Recursive_sequence.py
class Solution:
    def sequence(self, n):
        # code here
     i = 1
     c = 1
     s = 1
     p = 0
     mod = 1000000007
     while i <= n:
        k = i
        while k > 0:
            s = (s * c) % mod
            c += 1
            k -= 1
        p = (p + s) % mod
        s = 1
        i += 1
     return p